napari/_qt/_tests/test_qt_provide_theme.py

```diff
@@ -32,13 +33,37 @@ def napari_experimental_provide_theme():
     reg = napari_plugin_manager._theme_data["TestPlugin"]
     assert isinstance(reg["dark-test-2"], Theme)
 
+    # set the viewer theme
     viewer.theme = "dark-test-2"
 
     # triggered when theme was added
     mock_add_theme.assert_called()
     mock_remove_theme.assert_not_called()
 
     # now, lets unregister the theme
+    # We didn't set the setting, so there should be no warning
+    with pytest.warns(None):
+        napari_plugin_manager.unregister("TestPlugin")
+    mock_remove_theme.assert_called()
+
+    # reset mocks
+    mock_add_theme.reset_mock()
+    mock_remove_theme.reset_mock()
+
+    # re-register the theme
+    napari_plugin_manager.register(TestPlugin)
+    reg = napari_plugin_manager._theme_data["TestPlugin"]
+    assert isinstance(reg["dark-test-2"], Theme)
+
+    # this time, set the theme setting
+    get_settings().appearance.theme = "dark-test-2"
+
+    # triggered when theme was added
+    mock_add_theme.assert_called()
+    mock_remove_theme.assert_not_called()
+
+    # now, lets unregister the theme
+    # We did set the setting, so there should be a warning
```

Comment:
I do not like long tests. Maybe it could be moved to a new test function?

Reply:
OK, I will give it another look. I can't recall exactly what was being tested 🤣 
 I think toggle with added themes from plugin? so it should be able to be split into it's own test.

Reply:
I've made it two tests. One that checks for registering with no change of settings (just viewer.theme) and the second that checks with change of settings.